src/hotspot/share/gc/z/zRelocationSetSelector.cpp

```diff
@@ -42,50 +42,53 @@ ZRelocationSetSelectorGroupStats::ZRelocationSetSelectorGroupStats()
 
 ZRelocationSetSelectorGroup::ZRelocationSetSelectorGroup(const char* name,
                                                          ZPageType page_type,
-                                                         size_t page_size,
+                                                         size_t max_page_size,
                                                          size_t object_size_limit,
                                                          double fragmentation_limit)
   : _name(name),
     _page_type(page_type),
-    _page_size(page_size),
+    _max_page_size(max_page_size),
     _object_size_limit(object_size_limit),
     _fragmentation_limit(fragmentation_limit),
-    _page_fragmentation_limit((size_t)(page_size * (fragmentation_limit / 100))),
+    _page_fragmentation_limit((size_t)(_max_page_size * (_fragmentation_limit / 100))),
     _live_pages(),
     _not_selected_pages(),
     _forwarding_entries(0),
     _stats() {}
 
 bool ZRelocationSetSelectorGroup::is_disabled() {
   // Medium pages are disabled when their page size is zero
-  return _page_type == ZPageType::medium && _page_size == 0;
+  return _page_type == ZPageType::medium && !ZPageSizeMediumEnabled;
```

Comment:
The comment above described why the implementation looked like it did. Should it be updated?